File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
function findMaxAverage(nums, k) {
    let curSum = 0;

    for(let i = 0; i < k; i++){
        curSum += nums[i]
    }

    let maxAvg = curSum / k

    for(let i = k; i < nums.length; i++){
        curSum += nums[i];
        curSum -= nums[i - k]
        let avg = curSum / k;
        maxAvg = Math.max(maxAvg,avg)
    }
    return maxAvg
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        curSum = 0

        for i in range(k):
            curSum += nums[i]
        
        maxAvg = curSum / k
         
        for i in range(k,len(nums)):
            curSum += nums[i]
            curSum -= nums[i - k]
            avg = curSum / k
            maxAvg = max(avg,maxAvg)
        
        return maxAvg
